src/main/groovy/life/qbic/portal/sampletracking/web/presenters/SampleListPresenter.groovy

```diff
@@ -32,6 +32,21 @@ class SampleListPresenter implements SampleTrackingUpdateOutput, QuerySampleOutp
         viewModel.samples.removeAll {  sample -> (sample as Sample).code == sampleId }
     }
 
+    /**
+     * Signals that the update for process is finished and includes information on the samples updated successfully
+     * @param successfulCodes the sample codes for which the update was successful
+     * @param failedCodes the sample codes for which the update was unsuccessful
+     * @since 1.3.0
+     */
+    @Override
+    void updateFinished(Collection<String> successfulCodes, Collection<String> failedCodes) {
+        viewModel.successNotifications.add("Updated ${successfulCodes.size()} samples.")
+        if (failedCodes) {
+            viewModel.failureNotifications.add("Could not update\n" + failedCodes.join("\n\t"))
+        }
+        viewModel.samples.removeAll {  sample -> (sample as Sample).code in successfulCodes }
```

Comment:
```suggestion
        viewModel.samples.removeAll { Sample sample -> sample.code in successfulCodes }
```

Reply:
This does not work because entries in `samples` are of type Object. IDE complains :)

Reply:
Why is that? Shouldn't that be fixed?

Reply:
No would be a major refactor. This exceeds the scope of this PR

Reply:
I see..